ui/helpers/utils/util.js

```diff
@@ -584,3 +584,30 @@ export const sanitizeString = (value) => {
   const regex = /\u202E/giu;
   return value.replace(regex, '\\u202E');
 };
+
+/**
+ * This method checks current provider type and returns its string representation
+ *
+ * @param {*} provider
+ * @param {*} t
+ * @returns
+ */
+
+export const getNetworkNameFromProviderType = (providerConfig, t) => {
+  const providerName = providerConfig.type;
+
+  switch (providerName) {
+    case NETWORK_TYPES.MAINNET:
+      return t('mainnet');
+    case NETWORK_TYPES.GOERLI:
+      return t('goerli');
+    case NETWORK_TYPES.SEPOLIA:
+      return t('sepolia');
+    case NETWORK_TYPES.LINEA_TESTNET:
+      return t('lineatestnet');
+    case NETWORK_TYPES.LOCALHOST:
+      return t('localhost');
+    default:
+      return providerConfig.nickname || t('unknownNetwork');
+  }
+};
```

Comment:
It might be better to return strings to be internationalised from here and not pass `t`.